controllers/api.js

```diff
@@ -255,6 +255,19 @@ router.get('/getbtc', async function(req, res) {
   res.send([{ address }]);
 });
 
+router.get('/checkpayment/:payment_hash', async function(req, res) {
+  logger.log('/checkpayment', [req.id]);
+  let u = new User(redis, bitcoinclient, lightning);
+  await u.loadByAuthorization(req.headers.authorization);
+
+  if (!u.getUserId()) {
+    return errorBadAuth(res);
+  }
+
+  let paid = !!(await u.getPaymentHashPaid(req.params.payment_hash));
```

Comment:
A bit too late, I already merged, but well. This won't work right away, as `getPaymentHashPaid` checks only redis-cached flag. Someone has to set this flag first.
So what you need is to check `getPaymentHashPaid`. If it is not paid - use `User.lookupInvoice` and use that. Dont forget to set that flag via `setPaymentHashPaid` if it was paid after all.